### PR TITLE
Fix 18 JS type errors for TSC to output 683 TS/TSX errors

### LINKED ISSUE #5878: JavaScript static type errors
### Description

Follow-up to: Slack convo linked below
Fix our current 18 JS static type errors detected running our TypeScript compiler

### Screenshots

screenshot_2023-03-01_at_10 57 25-2

### To Reproduce

```sh
yarn tsc # locally
tsc # globally, nb: if installed, likely a diff ver so results may vary
```

### Expected behavior

No JS type errors

### **Technical Details**

Currently we rely on IDE's (e.g. VSCode) to flag type errors during development and bump into these in runtime

JavaScript and JSX files cannot parse types (yet, see 2 refs below)

Next: Address remaining TypeScript errors

### **References**

- https://consensys.slack.com/archives/G1L7H42BT/p1677193361291439
- https://consensys.slack.com/archives/C02U025CVU4/p1677256019459879

+ https://devblogs.microsoft.com/typescript/a-proposal-for-type-syntax-in-javascript
+ https://github.com/tc39/proposal-type-annotations

- https://github.com/MetaMask/metamask-mobile/issues/4605

-------------------------------------------------------------

**Severity**

- Is this tech debt? ✅

---

cc @Cal-L @gantunesr

### PR DESCRIPTION
**Pull request by leotm**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

Minimum required to see/tackle remaining 683 errors in 120 files in our TSC

Follow-up to: https://github.com/MetaMask/metamask-mobile/pull/5879
Fix: https://github.com/MetaMask/metamask-mobile/issues/5878

_After this PR merge https://github.com/MetaMask/metamask-mobile/pull/5882 to reduce to 556 errors in 87 files_

Then i can begin refactoring to TS (in similar manner to @gantunesr's doing in https://github.com/MetaMask/KeyringController/pull/202)

cc @Cal-L

**Screenshots/Recordings**

In aforementioned links above ^

**Issue**

Progresses https://github.com/MetaMask/metamask-mobile/issues/5878

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

File: app/core/Engine.ts [hunk 1]
```diff
@@ -801,9 +800,9 @@ class Engine {
 
     // Recreate imported accounts
     if (importedAccounts) {
-      for (let i = 0; i < importedAccounts.length; i++) {
+      for (const importedAccount of importedAccounts) {
         await KeyringController.importAccountWithStrategy('privateKey', [
-          importedAccounts[i],
+          importedAccount,
         ]);
       }
```
Comment (leotm) [inline app/core/Engine.ts:807]: simpler than https://github.com/MetaMask/metamask-mobile/pull/5879/files#r1126742680

File: app/core/Engine.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-shadow */
```
Comment (leotm) [inline app/core/Engine.ts:1]: see: https://github.com/MetaMask/metamask-mobile/pull/5879#issuecomment-1453943474 and https://github.com/MetaMask/metamask-mobile/pull/5879#issuecomment-1456412685

**Comment by Cal-L** [comment]
After running `yarn tsc`, I'm seeing a bunch of errors from `node_modules` and from the app. Any ideas what I might be doing wrong?

**Comment by leotm** [comment]
> After running `yarn tsc`, I'm seeing a bunch of errors from `node_modules` and from the app. Any ideas what I might be doing wrong?

yep that's expected ^ so this pr fixes our
- https://github.com/MetaMask/metamask-mobile/issues/5878 (initial 18 JS type errors atm, running `yarn tsc` in `main`)

which enables TSC to emit our 683 TS/TSX type errors lurking behind 😅

so next step is to tackle these ^ just created an issue for us here
- https://github.com/MetaMask/metamask-mobile/issues/5993

**Comment by leotm** [comment]
after merging current pr, this pr reduces the `node_modules` errors by a lot
- https://github.com/MetaMask/metamask-mobile/pull/5882

so think you're seeing the screenshots in the issue
- https://github.com/MetaMask/metamask-mobile/issues/5881

but i can't seem to figure why the last few `node_modules` errors remain there 🤔

**Review by Cal-L — APPROVED** [review]
Left a small comment. LGTM

File: app/core/Engine.ts [hunk 3]
```diff
@@ -482,7 +484,7 @@ class Engine {
       transaction.configure({ sign: keyring.signTransaction.bind(keyring) });
       this.controllerMessenger.subscribe(
         AppConstants.NETWORK_STATE_CHANGE_EVENT,
-        (state: { network: string, providerConfig: { chainId: any } }) => {
+        (state: { network: string; providerConfig: { chainId: any } }) => {
```
[the remark this answers is not in the document]
Comment (leotm) [reply app/core/Engine.ts:487]: sorry been having internet outages, great question ^

it's coming from our inferred Prettier parser (TypeScript by default)

Screenshot 2023-03-21 at 13 07 06

as a test overrides to JSON

Screenshot 2023-03-21 at 13 08 22

our rules no longer detected

[the remark this answers is not in the document]
Comment (leotm) [reply app/core/Engine.ts:487]: so definitely a Prettier rule we can tweak as a follow-up

Screenshot 2023-03-21 at 13 01 41

**Review by gantunesr — APPROVED** [review]
LGTM

**Pull request merged**